libtiledbvcf/src/stats/variant_stats_reader.h

```diff
@@ -102,6 +103,46 @@ class AFMap {
            pos_map.first;
   }
 
+  /**
+   * @brief Compute the Allele Frequency for an allele at the given position,
+   * accounting for the full population of the dataset
+   *
+   * @param pos Position of the allele
+   * @param allele Allele value
+   * @param num_samples Number of samples in the entire dataset
+   * @return float Allele Frequency
+   */
+  float af(
+      uint32_t pos,
+      const std::string& allele,
+      std::vector<std::vector<char>>::size_type num_samples) {
+    // calculate af = ac / an
+    std::unordered_map<
+        uint32_t,
+        std::pair<int, std::unordered_map<std::string, int>>>::const_iterator
+        pos_map_iterator = ac_map_.find(pos);
+
+    // Return -1.0 if the allele was not called
+    if (pos_map_iterator == ac_map_.end()) {
+      return -1.0;
+    }
+
+    const std::pair<int, std::unordered_map<std::string, int>>& pos_map =
+        pos_map_iterator->second;
+
+    // We don't know that allele was called in this sample. Ask nicely for the
+    // allele count.
+    decltype(pos_map.second)::const_iterator next_allele =
+        pos_map.second.find(allele);
+
+    // First multiply by 1.0 to force a float type, then look up AC from the
+    // above iterator. Substitute 0 for the AC value if it is absent in pos_map.
+    // Divide by AN (pos_map.first) to get AF.
+    return 1.0 *
+           (next_allele == pos_map.second.end() ? 0 : next_allele->second) /
```

Comment:
This is where `if/else` is more readable than a ternary operator.

Reply:
This is also where the code is not DRY; would it make sense for me to dedup the `af()` methods?

Reply:
Yes, it's worth looking at deduping the `af()` methods.